core/unit_test/TestTeamBasic.hpp

```diff
@@ -114,6 +114,83 @@ TEST(TEST_CATEGORY, team_reduce_large) {
   }
 }
 
+/*! \brief Test passing an aggregate to Kokkos::single in a parallel_for with
+           team policy
+*/
+template <typename ExecutionSpace>
+struct TestTeamForAggregate {
+  using range_policy_t = Kokkos::RangePolicy<ExecutionSpace>;
+  using team_policy_t  = Kokkos::TeamPolicy<ExecutionSpace>;
+  using member_t       = typename team_policy_t::member_type;
+  using memory_space   = typename ExecutionSpace::memory_space;
+  using results_type   = Kokkos::View<double*, memory_space>;
+
+  static constexpr double INIT_VALUE   = -1.0;
+  static constexpr double EXPECT_VALUE = 1.0;
+
+  struct Agg {
+    double d;
+  };
+  results_type results_;
+
+  TestTeamForAggregate(const size_t size) : results_("results", size) {}
+  TestTeamForAggregate() : TestTeamForAggregate(0) {}
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const member_t& t) const {
+    Agg lagg;
+    lagg.d = INIT_VALUE;
+    Kokkos::single(
+        Kokkos::PerTeam(t), [&](Agg& myAgg) { myAgg.d = EXPECT_VALUE; }, lagg);
+    size_t i = t.league_rank() * t.team_size() + t.team_rank();
+    if (i < results_.size()) {
+      results_(i) = lagg.d;
+    }
+  }
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const int i, int& lNumErrs) const {
+    if (EXPECT_VALUE != results_(i)) {
+      ++lNumErrs;
+    }
+  }
+
+  static void run() {
+    int minTeamSize = 1;
+    /* OpenMPTarget hard-codes 32 as the minimum size
+       FIXME OPENMPTARGET
+    */
+#ifdef KOKKOS_ENABLE_OPENMPTARGET
+    if constexpr (std::is_same<ExecutionSpace,
+                               Kokkos::Experimental::OpenMPTarget>::value) {
+      minTeamSize = 32;
+    }
+#endif
+
+    int maxTeamSize;
+    {
+      TestTeamForAggregate test;
+      maxTeamSize = team_policy_t(1, minTeamSize)
+                        .team_size_max(test, Kokkos::ParallelForTag());
+    }
+
+    for (int teamSize = minTeamSize; teamSize <= maxTeamSize; teamSize *= 2) {
+      for (int problemSize : {1, 100, 10'000, 1'000'000}) {
+        const int leagueSize = (problemSize + teamSize - 1) / teamSize;
+        TestTeamForAggregate test(problemSize);
+        Kokkos::parallel_for(team_policy_t(leagueSize, teamSize), test);
+        int numErrs = 0;
+        Kokkos::parallel_reduce(range_policy_t(0, problemSize), test, numErrs);
+        EXPECT_EQ(numErrs, 0);
```

Comment:
Consider printing `teamSize` and `problemSize`
```suggestion
        EXPECT_EQ(numErrs, 0) << " teamSize=" << teamSize << " problemSize=" << problemSize;
```